### PR TITLE
[AIRFLOW-1300] Enable table creation with TBLPROPERTIES in HiveCLIHook

### PR DESCRIPTION
**Pull request by krishnabhupatiraju**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:

- [AIRFLOW-1300](https://issues.apache.org/jira/browse/AIRFLOW-1300)

### CODE REVIEW

File: airflow/hooks/hive_hooks.py [hunk 1]
```diff
@@ -383,8 +384,13 @@ def load_file(
                 hql += "PARTITIONED BY ({pfields})\n"
             hql += "ROW FORMAT DELIMITED\n"
             hql += "FIELDS TERMINATED BY '{delimiter}'\n"
-            hql += "STORED AS textfile;"
+            hql += "STORED AS textfile\n"
+            if tblproperties:
```
Comment (aoen) [inline airflow/hooks/hive_hooks.py]: You don't need the if statement.

File: airflow/hooks/hive_hooks.py [hunk 2]
```diff
@@ -383,8 +384,13 @@ def load_file(
...
+                tprops = ", ".join(
+                    ["'{0}'='{1}'".format(k, v) for k, v in tblproperties.items()])
+                hql += "TBLPROPERTIES({tprops})\n"
         hql = hql.format(**locals())
```
Comment (aoen) [inline airflow/hooks/hive_hooks.py]: Put this above the hql.format.

**Comment by bolkedebruin** [comment]
Tblproperties? Any reason not to use the more pythonic table_properties?

File: airflow/hooks/hive_hooks.py [hunk 3]
```diff
@@ -343,7 +343,8 @@ def load_file(
             create=True,
             overwrite=True,
             partition=None,
-            recreate=False):
+            recreate=False,
+            tblproperties={}):
```
Comment (saguziel) [inline airflow/hooks/hive_hooks.py]: I'm gonna recommend not doing tblproperties={}, better off doing None. The reason is that the same dictionary is shared across all invocations which could cause problems later.

File: airflow/hooks/hive_hooks.py (hunk 1 above)
Comment (saguziel) [reply airflow/hooks/hive_hooks.py]: I'll recommend checking `if tblproperties is not None` isntead, and changing tblproperties above to None

**Comment by saguziel** [comment]
You should also change mssql_to_hive and mysql_to_hive

**Comment by saguziel** [comment]
@bolkedebruin the field is actually called TBLPROPERTIES in Hive (https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL)

**Comment by saguziel** [comment]
Also, please change the formatting to match the template for the PR body

File: airflow/hooks/hive_hooks.py [hunk 4]
```diff
@@ -343,7 +343,8 @@ def load_file(
             create=True,
             overwrite=True,
             partition=None,
-            recreate=False):
+            recreate=False,
+            table_properties={}):
```
Comment (saguziel) [inline airflow/hooks/hive_hooks.py]: change to None please

Comment (saguziel) [reply airflow/hooks/hive_hooks.py]: Also, change this back to tblproperties since that is what it is called in Hive

Comment (saguziel) [reply airflow/hooks/hive_hooks.py]: It's confusing if the name is changed

File: airflow/hooks/hive_hooks.py [hunk 5]
```diff
@@ -383,7 +393,12 @@ def load_file(
                 hql += "PARTITIONED BY ({pfields})\n"
             hql += "ROW FORMAT DELIMITED\n"
             hql += "FIELDS TERMINATED BY '{delimiter}'\n"
-            hql += "STORED AS textfile;"
+            hql += "STORED AS textfile\n"
+            if table_properties:
```
Comment (saguziel) [inline airflow/hooks/hive_hooks.py]: if tblproperties is not None

**Comment by krishnabhupatiraju** [comment]
@saguziel @aoen @bolkedebruin made requested changes. please take another look.

**Comment by saguziel** [comment]
Looks like tests are broken in master too, query renders correctly in stack trace
```
root: INFO: DROP TABLE IF EXISTS test_mysql_to_hive;
   CREATE TABLE IF NOT EXISTS test_mysql_to_hive (
   org_year INT,
       baby_name STRING,
       rate DOUBLE,
       sex STRING)
   ROW FORMAT DELIMITED
   FIELDS TERMINATED BY ','
   STORED AS textfile
   TBLPROPERTIES('test_property'='test_value')
   ;
```

**Comment by krishnabhupatiraju** [comment]
@saguziel The tests are now successful

**Pull request closed without merging**